From d3ba1059bde22124bd9f3a3827edbb838d58d6a0 Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Mon, 8 Dec 2025 09:24:07 -0500
Subject: [PATCH] create-item: Only configure audio device sink nodes in MONO

This allows multi channel mixing in the graph, even if MONO is enabled.
---
 src/scripts/node/create-item.lua | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/scripts/node/create-item.lua b/src/scripts/node/create-item.lua
index e395a9f3..259ebc83 100644
--- a/src/scripts/node/create-item.lua
+++ b/src/scripts/node/create-item.lua
@@ -16,6 +16,7 @@ items = {}
 function configProperties (node)
   local properties = node.properties
   local media_class = properties ["media.class"] or ""
+  local factory_name = properties ["factory.name"] or ""
 
   -- ensure a media.type is set
   if not properties ["media.type"] then
@@ -40,6 +41,7 @@ function configProperties (node)
   properties ["item.features.control-port"] =
       Settings.get_boolean ("node.features.audio.control-port")
   properties ["item.features.mono"] =
+      (factory_name == "api.alsa.pcm.sink" or factory_name == "api.bluez5.a2dp.sink") and
       Settings.get_boolean ("node.features.audio.mono")
   properties ["node.id"] = node ["bound-id"]
 
-- 
2.52.0

